### PR TITLE
ENG-1418: removing wrong condition

### PR DESCRIPTION
**Pull request by gReis89**
This feature was tested and it was working fine during tests done on [ENG-1418](https://jira.entando.org/browse/ENG-1418) ticket, but for some reason, when the same test was done on [ENG-1356](https://jira.entando.org/browse/ENG-1356), the modal was not being opened.

After testing again locally, I was able to reproduce the issue, and the root cause was an condition I added to prevent the fetch to happen multiple times by comparing the prevProps `dumpData` with the new prop. For some reason, it was working during ENG-1418 tests, but now this condition was never match, so I removed it.

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **greis89**

If they are not already assigned, you can assign the PR to them by writing `/assign @greis89` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Review by ichalagashvili — APPROVED** [review]

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/app-builder/pull/904#pullrequestreview-524326327" title="Approved">ichalagashvili</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fapp-builder).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando/app-builder/blob/master/OWNERS)~~ [ichalagashvili]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
